xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-api/src/main/java/org/xwiki/notifications/script/NotificationScriptEventHelper.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.notifications.script;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.xwiki.bridge.DocumentAccessBridge;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.eventstream.Event;
+import org.xwiki.eventstream.EventStatus;
+import org.xwiki.eventstream.EventStatusManager;
+import org.xwiki.eventstream.internal.DefaultEvent;
+import org.xwiki.eventstream.internal.DefaultEventStatus;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+import org.xwiki.notifications.CompositeEvent;
+import org.xwiki.notifications.CompositeEventStatus;
+import org.xwiki.notifications.CompositeEventStatusManager;
+
+/**
+ * Helper for event related operations on the {@link NotificationScriptService}.
+ * This helper limits the number of imports in {@link NotificationScriptService}.
+ *
+ * @version $Id$
+ * @since 9.7RC1
+ */
+@Component(roles = NotificationScriptEventHelper.class)
+@Singleton
+public class NotificationScriptEventHelper
```

Comment:
BTW this is not a nice name and not a nice way to implement a component IMO. Looks like there's a design issue (whenever you have a *Util class there's a design issue somewhere).